hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/infra/IterableStorageManager.java

```diff
@@ -51,29 +61,200 @@ public IterableStorageManager() {
+     *
+     * @param store Contract storage store
+     * @param newValue The new value for the slot
+     * @param contractNumber The contract number under consideration
+     * @param key The slot key to update
+     */
+    private void updateAccessedValue(
+            @NonNull final ContractStateStore store,
+            @NonNull Bytes newValue,
+            long contractNumber,
+            @NonNull final Bytes key) {
+        try {
+            Objects.requireNonNull(store);
+            Objects.requireNonNull(newValue);
+            Objects.requireNonNull(key);
+            // Look up the existing slot value for modification
+            final var slotKey = newSlotKeyFor(contractNumber, key);
+            final var slotValue = newSlotValueFor(store, true, slotKey, "Missing key ", key);
+
+            // Create updated new slot value and put into the store
+            final var newSlotValue = slotValue.copyBuilder().value(newValue).build();
+            store.putSlot(new SlotKey(contractNumber, key), newSlotValue);
+        } catch (Exception irreparable) {
+            log.error("Failed updating storage value for {}", key, irreparable);
+        }
+    }
+
+    /**
+     * Inserts the given key into the slot storage and into the linked list of storage for the given contract.
+     *
+     * @param store Contract storage store
+     * @param firstContractKey The first key in the linked list of storage for the given contract
+     * @param newValue The new value for the slot
+     * @param contractNumber The contract number under consideration
+     * @param newKey The slot key to insert
+     * @return the new first key in the linked list of storage for the given contract
+     */
+    @Nullable
+    private Bytes insertAccessedValue(
+            @NonNull final ContractStateStore store,
+            @Nullable final Bytes firstContractKey,
+            @NonNull final Bytes newValue,
+            long contractNumber,
+            @NonNull final Bytes newKey) {
+        try {
+            Objects.requireNonNull(store);
+            Objects.requireNonNull(newValue);
+            Objects.requireNonNull(newKey);
+            // Create new slot key and value and put into the store
+            final var newSlotKey = newSlotKeyFor(contractNumber, newKey);
+            final var newSlotValue = SlotValue.newBuilder()
+                    .value(newValue)
+                    .previousKey(null)
+                    .nextKey(firstContractKey)
+                    .build();
+            store.putSlot(newSlotKey, newSlotValue);
+            return newKey;
+        } catch (Exception irreparable) {
+            log.error("Failed link management when inserting {}", newKey, irreparable);
+        }
+        return firstContractKey;
+    }
+
+    @NonNull
+    private SlotKey newSlotKeyFor(long contractNumber, @NonNull final Bytes key) {
+        return new SlotKey(contractNumber, key);
+    }
+
+    @NonNull
+    private SlotValue newSlotValueFor(
```

Comment:
This isn't a `newSlotValueFor`, it's the _existing_ `SlotValue` ... (i.e., looking at the name `new...`)

Reply:
changed to `slotValueFor`